File: python/meteroid/models/metered_resolved_entitlement_value.py
# this file is @generated
import dataclasses
import typing as t
from decimal import Decimal

from ..serialization import BaseModel
from .billable_metric_id import BillableMetricId
from .reset_period import ResetPeriod


@dataclasses.dataclass
class MeteredResolvedEntitlementValue(BaseModel):
    enabled: bool

    metric_id: BillableMetricId

    reset_period: ResetPeriod

    limit: t.Optional[Decimal] = None

File: go/metered_resolved_entitlement_value.go
// this file is @generated
package meteroid

type MeteredResolvedEntitlementValue struct {
	Enabled bool `json:"enabled"`

	Limit *string `json:"limit,omitempty"`

	MetricId BillableMetricId `json:"metric_id"`

	ResetPeriod ResetPeriod `json:"reset_period"`
}
